jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ServletContextHandlerTest.java

```diff
@@ -619,7 +621,124 @@ public void testAddServletAfterStart() throws Exception
         assertThat("Response", response, containsString("Hello World"));
     }
+    
+    @Test
+    public void testServletRegistrationByClass() throws Exception
+    {
+        ServletContextHandler context = new ServletContextHandler();
+        context.setContextPath("/");
+        ServletRegistration reg = context.getServletContext().addServlet("test", TestServlet.class);
+        reg.addMapping("/test");
+
+        _server.setHandler(context);
+        _server.start();
+
+        StringBuffer request = new StringBuffer();
+        request.append("GET /test HTTP/1.0\n");
+        request.append("Host: localhost\n");
+        request.append("\n");
+
+        String response = _connector.getResponse(request.toString());
+        assertThat("Response", response, containsString("Test"));
+    }
+
+    @Test
+    public void testServletRegistrationByClassName() throws Exception
+    {
+        ServletContextHandler context = new ServletContextHandler();
+        context.setContextPath("/");
+        ServletRegistration reg = context.getServletContext().addServlet("test", TestServlet.class.getName());
+        reg.addMapping("/test");
+
+        _server.setHandler(context);
+        _server.start();
+
+        StringBuffer request = new StringBuffer();
+        request.append("GET /test HTTP/1.0\n");
+        request.append("Host: localhost\n");
+        request.append("\n");
+
+        String response = _connector.getResponse(request.toString());
+        assertThat("Response", response, containsString("Test"));
+    }
+    
+    @Test
+    public void testPartialServletRegistrationByName() throws Exception
+    {
+        ServletContextHandler context = new ServletContextHandler();
+        context.setContextPath("/");
+        ServletHolder partial = new ServletHolder();
+        partial.setName("test");
+        context.addServlet(partial, "/test");
+        
+        ServletRegistration reg = context.getServletContext().addServlet("test", TestServlet.class.getName());
+        assertNotNull(reg);
+        assertEquals(TestServlet.class.getName(), partial.getClassName());
+
+        _server.setHandler(context);
+        _server.start();
+
+        StringBuffer request = new StringBuffer();
+        request.append("GET /test HTTP/1.0\n");
+        request.append("Host: localhost\n");
+        request.append("\n");
+
+        String response = _connector.getResponse(request.toString());
+        assertThat("Response", response, containsString("Test"));
+    }
+    
+    @Test
+    public void testPartialServletRegistrationByClass() throws Exception
+    {
+        ServletContextHandler context = new ServletContextHandler();
+        context.setContextPath("/");
+        ServletHolder partial = new ServletHolder();
+        partial.setName("test");
+        context.addServlet(partial, "/test");
+        
+        ServletRegistration reg = context.getServletContext().addServlet("test", TestServlet.class);
```

Comment:
I comment here saying that the registration is providing the class name for the prior registration would be good

Reply:
Done